packages/amplify-codegen/src/amplify-plugin-index.js

```diff
@@ -14,7 +16,25 @@ async function handleAmplifyEvent(context, args) {
   context.print.info(`Received event args ${args}`);
 }
 
+/**
+ * Entry point for headless commands
+ * @param {any} context The amplify context object
+ * @param {string} headlessPayload The serialized payload from the platform
+ */
+async function executeAmplifyHeadlessCommand(context, headlessPayload) {
+  switch (context.input.command) {
+    case 'add':
+      await context.amplify.constructExeInfo(context);
+      context.exeInfo.inputParams[constants.Label] = JSON.parse(headlessPayload);
+      await add(context);
+      break;
+    default:
+      context.print.error(`Headless mode for ${context.input.command} codegen is not implemented yet`);
```

Comment:
The user may pass in a command that is erroneous, should the command be checked to see if it one that Amplify codegen supports? If it is, then it returns the error `Headless mode for ${context.input.command} codegen is not implemented yet ` and if not it says something like `${context.input.command} is not a valid codegen command`

Reply:
Thanks for the comment. `amplify-codegen` is one of the amplify plugins and command validation is conducted in the upper level of `amplify-cli`. You can find the source code here https://github.com/aws-amplify/amplify-cli/blob/99cb0133f38e0579869c71a88213eecb43bb7a8d/packages/amplify-cli/src/input-manager.ts#L96

Reply:
Ahh thanks. Makes total sense.